examples/hpc/dla-driver/src/tensor3.rs

```diff
@@ -0,0 +1,207 @@
+#![no_std]
+#![no_main]
+
+use alloc::vec::*;
+use ndarray::{Array, Array3};
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum Order3 {
+    CHW,
+    CWH,
+    HWC,
+    HCW,
+    WHC,
+    WCH,
+}
+
+impl Order3 {
+    fn into_position(self) -> [usize; 3] {
+        match self {
+            Order3::CHW => [0, 1, 2],
+            Order3::CWH => [0, 2, 1],
+            Order3::HWC => [1, 2, 0],
+            Order3::HCW => [1, 0, 2],
+            Order3::WHC => [2, 1, 0],
+            Order3::WCH => [2, 0, 1],
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Tensor3<T> {
+    data: Array3<T>,
+    order: Order3,
+}
+
+impl<T: Clone> Tensor3<T> {
+    // Creates a new Tensor3 with the specified dimensions, initial value, and order
+    pub fn new(
+        channels: usize,
+        height: usize,
+        width: usize,
+        initial_value: T,
+        order: Order3,
+    ) -> Self {
+        let data = Array::from_elem((channels, height, width), initial_value);
+        Tensor3 { data, order }
+    }
+
+    pub fn channels(&self) -> usize {
+        let dim_order: [usize; 3] = self.order.into_position();
+        let position = dim_order.iter().position(|&r| r == 0).unwrap();
```

Comment:
I would discourage use of `.unwrap()` in drivers. If you know **for sure** that this unwrap will **never** fail, you can use `unwrap_unchecked()`. I think that is appropriate in this case. Same for `height` and `width` functions.

Reply:
Okay, this makes sense. In this case there isn't a chance of failure, so I replaced `unwrap()` with `unwrap_unchecked()`.